backend/app/core/graceful_shutdown.py

```
@@ -0,0 +1,281 @@
+            The new count of active requests (minimum 0)
+        """
+        if not self._initialized or not self._redis_client:
+            return 0
+
+        try:
+            count = await self._redis_client.decr(ACTIVE_REQUESTS_KEY)
+            # Ensure count doesn't go below 0
+            if count < 0:
+                await self._redis_client.set(ACTIVE_REQUESTS_KEY, 0)
+                count = 0
+            return count
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)